### PR TITLE
FEAT: All Interfaces for the graphic side

### PR DESCRIPTION
**Pull request by Peralban**
# Pull Request: Add Comprehensive Documentation to Interfaces and Enums

This pull request adds extensive documentation to various interfaces and enums across the project.
The documentation includes:

- **Class-level descriptions**
- **Method-level descriptions**
- **Parameter details**

### Updated Files

---

### `Client/Interface/IFont.hpp`

- **Class Documentation**: `IFont` interface.
- **Documented Methods**: `GetFontDefault`, `LoadFont`.

---

### `Client/Interface/ISound.hpp`

- **Class Documentation**: `ISound` interface.
- **Documented Methods**:  
  `LoadSound`, `IsSoundReady`, `UpdateSound`, `UnloadSound`, `PlaySound`, `StopSound`,  
  `SetSoundVolume`, `GetSoundVolume`, `SetSoundPitch`, `GetSoundPitch`,  
  `SetSoundLoop`, `GetSoundLoop`.

---

### `Client/Interface/IMusic.hpp`

- **Class Documentation**: `IMusic` interface.
- **Documented Methods**:  
  `LoadMusic`, `IsMusicReady`, `PlayMusic`, `StopMusic`, `UnloadMusic`,  
  `SetMusicVolume`, `GetMusicVolume`, `SetMusicPitch`, `GetMusicPitch`,  
  `SetMusicLoop`, `GetMusicLoop`.

---

### `Client/Interface/IShader.hpp`

- **Class Documentation**: `IShader` interface.
- **Documented Methods**:  
  `LoadShaderFs`, `LoadShaderVs`, `SetShaderValue`, `SetShaderValueTexture`.

---

### `Client/Interface/IColor.hpp`

- **Struct Documentation**: `Color`.
- **Documented Members**:  
  `r`, `g`, `b`, `a`.

---

### `Client/Interface/IGraphic.hpp`

- **Class Documentation**: `IGraphic` interface.
- **Documented Methods**:  
  `Init`, `Update`, `Destroy`, `GetKey`, `Draw`, `Clear`,  
  `DrawTexts`, `HasText`, `PlaySounds`, `HasSound`,  
  `PlayMusics`, `HasMusic`, `Draw3DEntities`, `Has3DEntity`,  
  `Draw2DEntities`, `Has2DEntity`, `DrawTextures`, `HasTexture`,  
  `DrawShaders`, `HasShader`.

---

### `Client/Interface/IText.hpp`

- **Class Documentation**: `IText` interface.
- **Documented Methods**:  
  `LoadText`, `SetColor`, `SetSize`, `SetFont`, `SetPosition`, `DisplayText`.

---

### `Client/Interface/I3dEntities.hpp`

- **Class Documentation**: `I3DEntity` interface.
- **Enum Documentation**: `MATERIAL`.
- **Documented Methods**:  
  `Load3DEntity`, `LoadTexture`, `SetPosition`, `SetRotation`,  
  `SetScale`, `SetColor`.

---

### `Client/Interface/ITexture.hpp`

- **Class Documentation**: `ITexture` interface.
- **Documented Methods**:  
  `LoadTexture`, `AddTexture`, `SetTexturePosition`, `SetTextureSize`,  
  `SetTextureColor`, `SetTextureRotation`, `SetTextureScale`,  
  `SetTextureRect`, `SetTextureOrigin`.

---

### `Client/Interface/KeyMap.hpp`

- **Enum Documentation**: `KEY_MAP`.
- **Documented**:  
  All key mappings, including alphanumeric keys, function keys,  
  keypad keys, and Android key buttons.

---

### `Client/Interface/I2dEntities.hpp`

- **Class Documentation**: `I2DEntity` interface.
- **Documented Methods**:  
  `Load2DEntity`, `SetPosition`, `SetSize`, `SetColor`, `SetRotation`,  
  `SetScale`, `SetRect`, `SetOrigin`, `SetTexture`.

---

### Impact

This documentation enhances the **readability** and **maintainability** of the code by providing clear descriptions of the interfaces and methods. It will also help new developers understand the purpose and usage of each interface and enum more easily.

### CODE REVIEW

**Review by theophile-jr — APPROVED** [review]

**Review by mathieurobert1 — DISMISSED** [review]
Seems good, you can start to implement your raylib.so :100:

File: Client/Interface/I2dEntities.hpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        virtual void SetScale(float x, float y) = 0;
+
+        /**
+         * @brief Set the rectangle of the 2D entity.
+         * @param x The x-coordinate of the rectangle.
+         * @param y The y-coordinate of the rectangle.
+         * @param w The width of the rectangle.
+         * @param h The height of the rectangle.
+         */
+        virtual void SetRect(float x, float y, float w, float h) = 0;
```
Comment (Thomaltarix) [inline Client/Interface/I2dEntities.hpp:74]: I think you might add the direction of the rectangle, or the position of the x and y in the rectangle (Is it at the top left, the bottom left, the middle,  etc...) In the doxygen and maybe also in the doc 👍

File: Client/Interface/I3dEntities.hpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+         * @param path The path to the file containing the 3D entity.
+         * @return The loaded 3D entity.
+         */
+        virtual I3DEntity Load3DEntity(std::string const &path) = 0;
+
+        /**
+         * @brief Load textures for the 3D entity.
+         * @param textures A vector of pairs containing texture file paths and their corresponding material types.
+         */
+        virtual void LoadTexture(std::vector<std::pair<std::string, MATERIAL>>) = 0;
```
Comment (Thomaltarix) [inline Client/Interface/I3dEntities.hpp:57]: The name of the parameter is missing, and you should rename it `LoadTextures` coz as the parameter tells, you can load severaltextures

**Review by Thomaltarix — CHANGES_REQUESTED** [review]
somes changes about some methods and some docs, but overall that's very complete, great job

**Review by mathieurobert1 — APPROVED** [review]

**Review by Thomaltarix — APPROVED** [review]

**Review by AxelF44 — APPROVED** [review]

**Review by theophile-jr — APPROVED** [review]

**Pull request merged**